common/autoware_node/test/test_an_init_shutdown.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <autoware/node/node.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <lifecycle_msgs/msg/state.hpp>
+
+#include <gtest/gtest.h>
+
+#include <memory>
+
+class AutowareNodeInitShutdown : public ::testing::Test
+{
+public:
+  void SetUp() override { rclcpp::init(0, nullptr); }
+
+  void TearDown() override { rclcpp::shutdown(); }
+
+  rclcpp::NodeOptions node_options_an_;
+};
+
+TEST_F(AutowareNodeInitShutdown, NodeInitShutdown)
+{
+  std::shared_ptr<rclcpp::executors::SingleThreadedExecutor> executor;
+  executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
```

Comment:
It is better to declare and initialize at the same time.

```suggestion
  auto executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
```

Reply:
Thanks, addressed in https://github.com/autowarefoundation/autoware.core/pull/113/commits/de30f052a0b17bcc9de73ebda72ae2823d6fa34d